backend/chainlit/data/storage_clients.py

```diff
@@ -39,20 +40,68 @@ async def upload_file(self, object_key: str, data: Union[bytes, str], mime: str
 class S3StorageClient(BaseStorageClient):
     """
     Class to enable Amazon S3 storage provider
+
+    params:
+        bucket: Name of the S3 bucket
+        access_key: AWS access key
+        secret_key: AWS secret key
+        session_token: AWS session token
     """
-    def __init__(self, bucket: str):
+    def __init__(self, bucket: str, access_key: Optional[str] = None, secret_key: Optional[str] = None, session_token: Optional[str] = None):
```

Comment:
A quick trick here is to accept the s3 client itself as an optional arg, and if it's null make a default one. This allows true customization for the end user. For example, they may want to use the Stubber for unittests or pass in a botocore.config.Config to customize the behavior.